services/navigation/builtin/builtin.go

```diff
@@ -575,7 +581,75 @@ func (svc *builtIn) waypointIsDeleted() bool {
 func (svc *builtIn) Obstacles(ctx context.Context, extra map[string]interface{}) ([]*spatialmath.GeoObstacle, error) {
 	svc.mu.RLock()
 	defer svc.mu.RUnlock()
-	return svc.obstacles, nil
+
+	// get static geoObstacles
+	geoObstacles := svc.obstacles
+
+	for _, detector := range svc.motionCfg.ObstacleDetectors {
+		// get the vision service
+		visSvc, ok := svc.visionServicesByName[detector.VisionServiceName]
+		if !ok {
+			return nil, fmt.Errorf("vision service with name: %s not found", detector.VisionServiceName)
+		}
+
+		// get the detections
+		detections, err := visSvc.GetObjectPointClouds(ctx, detector.CameraName.Name, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// get current geo position of robot
+		gp, _, err := svc.movementSensor.Position(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// instantiate a localizer and use it to get our current position
+		localizer := motion.NewMovementSensorLocalizer(svc.movementSensor, gp, spatialmath.NewZeroPose())
+		currentPIF, err := localizer.CurrentPosition(ctx)
+		if err != nil {
+			return nil, err
+		}
+		// convert orientation of currentPIF to be left handed
+		heading := math.Mod(math.Abs(currentPIF.Pose().Orientation().OrientationVectorDegrees().Theta-360), 360)
+
+		// convert geo position into GeoPose
+		relativeToGeoPose := spatialmath.NewGeoPose(gp, heading)
+
+		// iterate through all detections and construct a geoObstacle to append
+		for i, detection := range detections {
+			// get the geometry's lat & lng along with its heading with respect to north as a left handed value
+			geomGeoPose := spatialmath.PoseToGeoPose(relativeToGeoPose, detection.Geometry.Pose())
+
+			// prefix the label of the geometry so we know it is transient and add extra info
+			label := "transient_" + strconv.Itoa(i) + "_" + detector.CameraName.Name
+			if detection.Geometry.Label() != "" {
+				label += "_" + detection.Geometry.Label()
+			}
+			detection.Geometry.SetLabel(label)
+
+			// determine the desired geometry pose
+			desiredPose := spatialmath.NewPose(
+				r3.Vector{0, 0, 0},
+				&spatialmath.OrientationVectorDegrees{
+					OZ: 1,
+					// convert the heading to be right handed
+					Theta: math.Mod(math.Abs(geomGeoPose.Heading()-360), 360),
```

Comment:
Can heading be outside of the -360 to +360 range?
If so, does this calculation work if the heading is outside the +-360 range?

Reply:
should not be possible since we do `math.Mod(..., 360)`